node/pkg/dal/app.go

```diff
@@ -4,43 +4,46 @@ import (
 	"context"
 	"errors"
 	"os"
+	"time"
 
 	"bisonai.com/orakl/node/pkg/dal/api"
 	"bisonai.com/orakl/node/pkg/dal/utils/initializer"
+	"bisonai.com/orakl/node/pkg/dal/utils/keycache"
 
 	"github.com/gofiber/fiber/v2"
 	"github.com/rs/zerolog/log"
 )
 
 func Run(ctx context.Context) error {
 	log.Debug().Msg("Starting DAL API server")
-	app, err := initializer.Setup(ctx)
-	if err != nil {
-		log.Error().Err(err).Msg("Failed to setup DAL API server")
-		return err
-	}
-	defer func() {
-		_ = app.Shutdown()
-	}()
+
+	keyCache := keycache.NewAPIKeyCache(1 * time.Hour)
+	keyCache.CleanupLoop(10 * time.Minute)
 
 	adminEndpoint := os.Getenv("ORAKL_NODE_ADMIN_URL")
 	if adminEndpoint == "" {
 		return errors.New("ORAKL_NODE_ADMIN_URL is not set")
 	}
 
-	err = api.Setup(ctx, adminEndpoint)
+	controller, err := api.Setup(ctx, adminEndpoint)
 	if err != nil {
 		log.Error().Err(err).Msg("Failed to setup DAL API server")
 		return err
 	}
-	api.ApiController.Start(ctx)
 
-	log.Debug().Str("Player", "DAL API").Msg("DAL API collector started")
+	app, err := initializer.Setup(ctx, controller, keyCache)
+	if err != nil {
+		log.Error().Err(err).Msg("Failed to setup DAL API server")
+		return err
+	}
+	defer func() {
+		_ = app.Shutdown()
```

Comment:
just out of curiosity, I've seen this pattern a lot in the code where the return value is assigned to a blank identifier, which is not used. Afaik the blank identifier is useful when a function returns multiple values and you are not interested in some of them. Is it still a good practice to use the blank identifier in this way?

Reply:
I don't think so, 
it didn't pass the linter when I didn't assign blank identifier, so it kind of became habit I think